### ISSUE TITLE
Get test coverage to 100%

### ISSUE THREAD

**Issue by jameswlane** [jameswlane/status-board#44]
We need to get our coverage to 100%, this will allow us to refactor and make code changes with a little more confidence then we would have now.

The following files need testing:
- [ ] src/job-dependencies/app/dependency.ts
- [x] src/job-dependencies/async/dependency.ts
- [ ] src/job-dependencies/easyRequest/dependency.ts
- [ ] src/job-dependencies/experimentalConfluence/dependency.ts
- [ ] src/job-dependencies/hipchat/dependency.ts
- [ ] src/job-dependencies/logger/dependency.ts
- [x] src/job-dependencies/moment/dependency.ts
- [ ] src/job-dependencies/postgreSQL/dependency.ts
- [x] src/job-dependencies/request/dependency.ts
- [ ] src/job-dependencies/storage/implementations/StorageFS.ts
- [ ] src/job-dependencies/storage/implementations/StorageRedis.ts
- [ ] src/job-dependencies/storage/dependency.ts
- [x] src/job-dependencies/underscore/dependency.ts
- [ ] src/job-dependencies/loader.ts
- [x] src/utilities/check-if-file-exists.ts
- [ ] src/webapp/routes/dashboard.ts
- [ ] src/webapp/routes/widget.ts
- [ ] src/webapp/logic.ts
- [ ] src/webapp/routes.ts
- [ ] src/webapp/server.ts
- [ ] src/config-manager.ts
- [ ] src/EventQueue.ts
- [ ] src/global-auth.ts
- [ ] src/helpers.ts
- [ ] src/hipchat.ts
- [ ] src/item-manager.ts
- [ ] src/job-initialiser.ts
- [ ] src/job-manager.ts
- [ ] src/logger.ts
- [ ] src/package-dependency-manager.ts
- [ ] src/path-resolver.ts
- [ ] src/Scheduler.ts
- [ ] src/status-board.ts
- [ ] src/stylus.ts
- [x] src/template-manager.ts

**Comment by jameswlane** [comment]
@SwamiKalagiri - You can drag the item under your name (Using the three stacked lines icon), when complete you can check it off and drag it under completed.

**Comment by jameswlane** [comment]
@SwamiKalagiri I created a [helper](https://github.com/jameswlane/status-board/blob/master/test/helpers/socket.ts) for testing Socket.IO in the application.

You can see the first set of working tests [here](https://github.com/jameswlane/status-board/blob/master/test/unit/EventQueue.test.ts#L20-L33).

I am still trying to figure out how to test [this](https://github.com/jameswlane/status-board/blob/master/src/EventQueue.ts#L20-L29) part of the code. I been trying to use the [Socket.IO client](https://github.com/jameswlane/status-board/blob/master/test/helpers/socket.ts) to connect to the server and then test what is emitted from the event. I have yet to get that to work. I could use a second pair of eyes on it when you have a second.

**Comment by jameswlane** [comment]
@SwamiKalagiri I added the test for Socket.IO [here](https://github.com/jameswlane/status-board/commit/6be394c2a85229bd0a5988f85ad56cd37fb0f4d0), let me know what you think of the implementation.

Also I noticed the only thing we are missing now in coverage is [this](https://github.com/jameswlane/status-board/blob/6be394c2a85229bd0a5988f85ad56cd37fb0f4d0/src/status-board.ts#L77)

```javascript
const config = options || {};
``` 
If status board receives no options it will give it a empty object. By looking at the code I think this is impossible to get to, due to we need to pass it a callback and the second argument is a callback. The only way to get this to work would be call it like this:
```javascript
statusBoard(undefined, () => {})
```
Long story short I think we can get rid of the `|| {}`, what do you think?

**Comment by SwamiKalagiri** [comment]
Thanks @jameswlane, implementation looks good to me. it seems
```
const config = options || {};
```
added  to avoid errors in [line 65](https://github.com/jameswlane/status-board/blob/master/src/status-board.ts#L65),[line 23](https://github.com/jameswlane/status-board/blob/master/src/status-board.ts#L23) and [line 52](https://github.com/jameswlane/status-board/blob/master/src/status-board.ts#L52).

i added [test](https://github.com/jameswlane/status-board/blob/master/test/unit/status-board.test.ts#L96-L106) to cover missing the code. I think we keep this for now. Once complete the test coverage then refactor code.

**Comment by jameswlane** [comment]
@SwamiKalagiri That works also 👍

**Comment by jameswlane** [comment]
@SwamiKalagiri we are close to the half way point.

**Comment by jameswlane** [comment]
@SwamiKalagiri we broke the half way point 20 files done 17 files left 🍾

**Comment by jameswlane** [comment]
@SwamiKalagiri I been working on a [Response helper](https://gist.github.com/jameswlane/e7e97d50697243038883f395f9213c30) it will be pushed when I am done testing `src/webapp/logic.ts`. I think this is a good base for building other express helpers to make the app easier to test.

What do you think?

**Comment by jameswlane** [comment]
Got board and did some [cleanup](https://github.com/jameswlane/status-board/commit/ae1ba1bfd1374f8d41c5ff95821e42b908eabf04)

Deleted 378 files totaling 78,443 lines of code
